Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +218,56 @@ function onInterval(): void {
     clients.ActiveClient.onInterval();
 }
 
+async function installVsix(vsixLocation: string, updateChannel: string): Promise<void> {
+    // Get the path to the VSCode command -- replace logic later when VSCode allows calling of
+    // workbench.extensions.action.installVSIX from TypeScript w/o instead popping up a file dialog
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsCodeScriptPath: string = function(platformInfo): string {
+        if (platformInfo.platform === 'win32') {
+            const vsCodeProcessPath: string = path.dirname(process.execPath);
+            const scriptName: string = path.basename(process.execPath) + '.cmd';
+            return path.join(vsCodeProcessPath, 'bin', scriptName);
+        } else {
+            const vsCodeProcessPath: string = path.basename(process.execPath);
+            const stdout: Buffer = execSync("which " + vsCodeProcessPath);
+            return stdout.toString().trim();
+        }
+    }(platformInfo);
+    if (!vsCodeScriptPath) {
+        telemetry.logLanguageServerEvent('findVsCodeScriptFailure');
+        return;
+    }
+
+    // Install the VSIX
+    const installCommand: string = vsCodeScriptPath + ' --install-extension ' + vsixLocation;
+    try {
+        if (updateChannel === 'Default') {
+            // Uninstall the current version, as the version to install is a previous version
+            const uninstallCommand: string = vsCodeScriptPath + ' --uninstall-extension ms-vscode.cpptools';
+            execSync(uninstallCommand);
+        }
+        execSync(installCommand);
+    } catch (error) {
+        telemetry.logLanguageServerEvent("installCommandFailure");
+    }
+}
+
+async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
+    // TODO query whether user has extensions.autoUpdate turned off?
```

Comment:
Nah.  If they opt in to the insiders channel, it is assumed that they want us to auto-update.  You can remove this comment.